2.0/m_joinpartspam.cpp

```diff
@@ -376,9 +371,39 @@ class ModuleJoinPartSpam : public Module
 
 			if (allowredirect && !jpss->redirect.empty())
 				Channel::JoinUser(user, jpss->redirect.c_str(), false, "", false, ServerInstance->Time());
-			return MOD_RES_DENY;
+			return true;
 		}
 
+		return false;
+	}
+
+	// After /CYCLE, check if the user should be blocked and kick if so
+	void OnPostCommand(const std::string& command, const std::vector<std::string>& parameters, LocalUser* user, CmdResult result, const std::string& original_line)
+	{
+		if (result == CMD_FAILURE || command != "CYCLE" || IS_OPER(user))
+			return;
+
+		Channel* chan = ServerInstance->FindChan(parameters[0]);
+		if (!chan || !chan->IsModeSet(jps.GetModeChar()))
+			return;
+
+		bool kick = BlockJoin(user, chan);
+		if (kick)
```

Comment:
Rather than kicking could you not check in OnPreCommand as to whether the user will be blocked?